### ISSUE TITLE
Support passing in custom ENV

### ISSUE THREAD

**Issue by zeisss** [coreos/third_party.go#3]
Can we add parameter to define ENV for the child `go build` call?

Background: I needed to crosscompile `nsproxy` for which I need to set `GOOS=linux` for the `go build` process. Passing it to the ouside `go run` fails obviously, as it would crosscompile the `third_party` as well.

**Comment by philips** [comment]
@ZeissS It should work. I simply override the GOPATH and GOBIN in third_party.go.

**Comment by jagregory** [comment]
I'm having issues with this as well. I'm trying to build etcd and it's failing, but the issue seems to be with third_party.

I'm on a Mac, and I'm cross compiling to Linux. If I run `GOOS=linux GOARCH=amd64 ./build` it fails with:

```
fork/exec /var/folders/m5/1fxblk2d409gk812l04r8rn00000gn/T/go-build555039057/command-line-arguments/_obj/exe/third_party: exec format error
```

Similar to what @ZeissS was saying, I believe what is happening is `go run` is using the `GOOS` and `GOARCH` environment variables I'm setting causing third_party to be compiled as Linux/AMD64; then when third_party actually tries to run install for etcd it fails because there's an architecture mismatch.

If I edit `third_party.go` and hard-code the `GOOS` and `GOARCH` variables in `main` and just execute `./build` then it works as expected.

``` golang
// third_party manages GOPATH, no one else
os.Setenv("GOPATH", thirdPartyDir())
os.Setenv("GOBIN", binDir())
os.Setenv("GOOS", "linux")
os.Setenv("GOARCH", "amd64")
```

**Comment by philips** [comment]
This is fixed via #5 but I will just convert nsproxy to use goven as I have done with etcd and etcdctl.

**Comment by philips** [comment]
@ZeissS Or if you can get to it first that would be great too.
